src/fairchem/core/modules/element_references.py

```diff
@@ -0,0 +1,229 @@
+            state_dict = torch.load(file)
+        elif extension == ".npz":
+            state_dict = {}
+            with np.load(file) as values:
+                # legacy linref files
+                if "coeff" in values:
+                    state_dict["element_references"] = torch.tensor(values["coeff"])
+                else:
+                    state_dict["element_references"] = torch.tensor(
+                        values["element_references"]
+                    )
+        else:
+            raise RuntimeError(
+                f"Element references file with extension '{extension}' is not supported."
+            )
+
+    if "element_references" not in state_dict:
+        raise RuntimeError("Unable to load linear element references!")
+
+    return LinearReference(element_references=state_dict["element_references"])
+
+
+def fit_linear_references(
+    targets: list[str],
+    dataset: Dataset,
+    batch_size: int,
+    num_batches: int | None = None,
+    num_workers: int = 1,
+    max_num_elements: int = 118,
+    driver: str | None = None,
+    shuffle: bool = True,
+    seed: int = 0,
+) -> dict[str, LinearReference]:
+    """Fit a set linear references for a list of targets using a given number of batches.
+
+    Args:
+        targets: list of target names
+        dataset: data set to fit linear references with
+        batch_size: size of batch
+        num_batches: number of batches to use in fit. If not given will use all batches
+        num_workers: number of workers to use in data loader
+        max_num_elements: max number of elements in dataset. If not given will use an ambitious value of 118
+        driver: backend used to solve linear system. See torch.linalg.lstsq docs.
+        shuffle: whether to shuffle when loading the dataset
+        seed: random seed used to shuffle the sampler if shuffle=True
+
+    Returns:
+        dict of fitted LinearReference objects
+    """
+    data_loader = DataLoader(
+        dataset,
+        batch_size=batch_size,
+        shuffle=shuffle,
+        collate_fn=partial(data_list_collater, otf_graph=True),
+        num_workers=num_workers,
+        pin_memory=True,
+        generator=torch.Generator.manual_seed(seed),
+    )
+
+    num_batches = num_batches if num_batches is not None else len(data_loader)
+    if num_batches > len(data_loader):
+        logging.warning(
+            f"The give num_batches {num_batches} is larger than total batches of size {batch_size} in dataset. "
+            f"Will ignore num_batches and use the whole dataset"
+        )
+        num_batches = len(data_loader)
+
+    max_num_elements += 1  # + 1 since H starts at index 1
+    # solving linear system happens on CPU, which allows handling poorly conditioned and
+    # rank deficient matrices, unlike torch lstsq on GPU
+    composition_matrix = torch.zeros(
+        num_batches * batch_size,
+        max_num_elements,
+    )
+
+    # This only works with scalar properties
```

Comment:
Let's throw an assert/error here to tell this to the user so they don't specify that in the config. Maybe you do that elsewhere